apache/drill
# DRILL-8322: Add a list of scanned plugin names to the query profile

# [DRILL-8322](https://issues.apache.org/jira/browse/DRILL-8322): Add a list of scanned plugin names to the query profile

## Description

A useful piece of information about a query is the set of plugins that it attempted to scan. While some such information is present in the physical plan text in the query profile, an easy to parse, sorted and deduplicated list of plugin names is not. This PR adds such a list.

Query profile fragment including the new scannedPlugins array
```
    "user": "alice",
    "optionsJson": "[ {\n  \"kind\" : \"BOOLEAN\",\n  \"accessibleScopes\" : \"ALL\",\n  \"name\" : \"metastore.enabled\",\n  \"bool_val\" : false,\n  \"scope\" : \"QUERY\"\n} ]",
    "planEnd": 1664464547555,
    "queueWaitEnd": 1664464547555,
    "totalCost": 260,
    "queueName": "Unknown",
    "queryId": "1cca495d-6dcd-4620-8027-0cf056e3b5bc",
    "scannedPlugins": [
        "cp",
        "dfs",
        "sys"
    ]
```

## Documentation
~~Add an entry to https://drill.apache.org/docs/query-profile-column-descriptions/~~

## Testing
View old query profiles from before this change in the web UI.
View new query profiles created after this change in web UI, check scannedPlugins array for

- SELECT against cp
- SELECT against dfs
- CTAS from cp to dfs
- SELECT against cp + dfs + sys

## Notes
I didn't turn up any unit testing of the contents of query profiles when I looked.
I haven't added anything to Profile detail page in the Drill web UI to present the new list of scanned plugins visually.
